### PR TITLE
Steam oauth login

### PR DESCRIPTION
**Pull request by harrisonlingren**
For #4 & #7

### CODE REVIEW

File: SteamStats.py [hunk 1]
```diff
@@ -1,18 +1,70 @@
...
+    return oid.try_login('https://steamcommunity.com/openid')
+
+@oid.after_login
+def create_or_login(auth):
+    _steam_re = re.compile('steamcommunity.com/openid/id/(.*?)$')
+    print('response: ' + str(auth))
+    match = _steam_re.search(auth.identity_url)
+    g.user = {}
+    steamid = match.group(1)
+    print('user: ' + str(steamid))
```
Comment (harrisonlingren) [inline SteamStats.py]: remove debugging

File: SteamStats.py [hunk 2]
```diff
@@ -1,18 +1,70 @@
...
+# login flow
+@app.route('/login')
+@oid.loginhandler
+def login():
+    return oid.try_login('https://steamcommunity.com/openid')
+
+@oid.after_login
+def create_or_login(auth):
+    _steam_re = re.compile('steamcommunity.com/openid/id/(.*?)$')
+    print('response: ' + str(auth))
```
Comment (harrisonlingren) [inline SteamStats.py]: remove debugging

File: SteamStats.py [hunk 3]
```diff
@@ -1,18 +1,70 @@
...
+
+oid = OpenID(app)
 
 @app.route('/')
 def index():
-    return render_template('index.jinja2')
+    print(session)
+    if 'user_id' in session:
+        user_logged_in = True
+        #print('logged in? ' + str(session['user_id']))
```
Comment (harrisonlingren) [inline SteamStats.py]: remove debugging

**Review by harrisonlingren — COMMENTED** [review]
Review comments

**Review by momm3 — APPROVED** [review]
lgtm

**Pull request merged**